diff --git a/source/session/tracking/vspace.d b/source/session/tracking/vspace.d
index d39f8d0..78d732c 100644
--- a/source/session/tracking/vspace.d
+++ b/source/session/tracking/vspace.d
@@ -339,6 +339,7 @@ public:
         if (sources.length > 1) {
             float count = 1;
             foreach(ref source; sources[1..$]) {
+                if(!source) continue;
                 if (name in source.getBones) {
                     count += 1;
 
